great_expectations/dataset/base.py

```diff
@@ -3010,3 +3010,152 @@ def expect_column_kl_divergence_to_be_less_than(self,
 
         """
         raise NotImplementedError
+
+    ### Column pairs ###
+
+    def expect_column_pair_values_to_be_equal(self,
+        column_A,
+        column_B,
+        keep_missing="either",
+        output_format=None, include_config=False, catch_exceptions=None
+    ):
+        """
+        Expect the values in column A to be the same as column B.
+
+        Args:
+            column_A (str): The first column name
+            column_B (str): The second column name
+
+        Keyword Args:
+            keep_missing (str): "either" or "both"
+
+        Other Parameters:
+            output_format (str or None): \
+                Which output mode to use: `BOOLEAN_ONLY`, `BASIC`, `COMPLETE`, or `SUMMARY`.
+                For more detail, see :ref:`output_format <output_format>`.
+            include_config (boolean): \
+                If True, then include the expectation config as part of the result object. \
+                For more detail, see :ref:`include_config`.
+            catch_exceptions (boolean or None): \
+                If True, then catch exceptions and include them as part of the result object. \
+                For more detail, see :ref:`catch_exceptions`.
+            meta (dict or None): \
+                A JSON-serializable dictionary (nesting allowed) that will be included in the output without modification. \
+                For more detail, see :ref:`meta`.
+
+        Returns:
+            A JSON-serializable expectation result object.
+
+            Exact fields vary depending on the values passed to :ref:`output_format <output_format>` and
+            :ref:`include_config`, :ref:`catch_exceptions`, and :ref:`meta`.
+
+        Notes:
+            Missing values are tricky in `column_pair` expectations. By default, `expect_column_pair_values_to_be_equal` compares
+            values from column A and column B unless *both* values are missing. This is `keep_missing="either".` That means that
+            missing values will be compared to non-missing values and found to be not equal. You can override this behavior by
+            setting `keep_missing="both"`.
+
+        """
+        raise NotImplementedError
+
+    def expect_column_pair_values_A_to_be_greater_than_B(self,
+        column_A,
+        column_B,
+        or_equal=None,
+        parse_strings_as_datetimes=None,
+        allow_cross_type_comparisons=None,
+        keep_missing="either",
```

Comment:
needs update to new ignore_row_if name.

Also: eventually we may want to remove parse_strings_as_datetimes option from base, since in a typed database system it's not clear we want to support in-expectation type changes. Not for this PR.